src/shared/clients/ecsClient.ts

```diff
@@ -5,70 +5,69 @@
 
 import { ECS } from 'aws-sdk'
 import { ext } from '../extensionGlobals'
+import { getLogger } from '../logger'
 import { ClassToInterfaceType } from '../utilities/tsUtils'
 
 export type EcsClient = ClassToInterfaceType<DefaultEcsClient>
+
+export type ServicesAndToken =  {
+    services: ECS.Service[],
+    nextToken?: string
+}
+
+export type ClustersAndToken = {
+    clusters: ECS.Cluster[],
+    nextToken?: string
+}
+
+const MAX_RESULTS = 50
 export class DefaultEcsClient {
     public constructor(public readonly regionCode: string) {}
 
-    public async *listClusters(): AsyncIterableIterator<string> {
+    public async listClusters(nextToken?: string): Promise<ClustersAndToken> {
         const sdkClient = await this.createSdkClient()
-        const request: ECS.ListClustersRequest = {}
-        do {
-            const response = await this.invokeListClusters(request, sdkClient)
-            if (response.clusterArns) {
-                yield* response.clusterArns
+        const clusterArnList = await sdkClient.listClusters({maxResults: MAX_RESULTS, nextToken}).promise()
+        try {
+            const clusterResponse =  await sdkClient.describeClusters({ clusters: clusterArnList.clusterArns}).promise()
+            const response: ClustersAndToken = {
+                clusters: clusterResponse.clusters!,
+                nextToken: clusterArnList.nextToken
             }
-            request.nextToken = response.nextToken
-        } while (request.nextToken)
-    }
-
-    public async *listServices(cluster: string): AsyncIterableIterator<string> {
-        const sdkClient = await this.createSdkClient()
-        const request: ECS.ListServicesRequest = {
-            cluster,
+            return response
+        } catch (error) {
+            getLogger().error('ecs: Failed to list clusters: %s', error)
+            throw error
```

Comment:
Nit: Logging and re-throwing the error as-is is usually a poor practice, though some of our other wrapped clients are doing this as well? Pass-through error catching can be useful if we wanted to add a generic logging statement to all wrapped clients, but in this case we're manually doing it for every single call.